docs/developers/data/abi.md

```diff
@@ -279,3 +279,576 @@ Similarly, [enums](/developers/data/custom-types#custom-enums) are defined by:
         - Struct-like variants, with named fields.
 
 You can read more about Rust enums [here](https://doc.rust-lang.org/book/ch06-01-defining-an-enum.html).
+
+---
+
+[comment]: # (mx-context-auto)
+
+## Custom ABI exports
+
+### Overview
+
+Even though the `#[derive(TypeAbi)]` annotation exports all the data types used in the contract along with the type descriptions, we have implemented a manual way to export any kind of data type (used inside the contract or not) in the ABI file in order to fetch it easier from other services (e.g. use-case: the developer can define the type of the attributes expected for a specific ESDT, fetch it from abi and use it easily in the frontend for a more specific result).
+
+Starting with the framework version `0.44`, developers can use the new trait annotation `#[esdt_attribute("name", Type)]` in order to export any data types in the ABI file. 
+
+:::important Important
+Please note that it is a `trait annotation`, meaning that it can only be used at trait level along with other `#[multiversx_sc]` annotation. Using it at endpoint level or at trait level outside `multiversx_sc` environment will not work.
+:::
+
+A new field called `esdtAttributes` was added to the ABI file, where developers can find the structs (name, type) exported using the `esdt_attribute` trait annotation. Additionally, each `esdt_attribute` will create a new json file with the name given by the developer (followed by `.esdt-abi`) and containing its exported structs (names, types and descriptions). 
+
+### Basic types
+
+Let's take the `Adder` contract as an example and try out the new annotation.
+
+__adder.rs:__
+```rust
+#[multiversx_sc::contract]
+#[esdt_attribute("testBasic", BigUint)]
+pub trait Adder {
```

Comment:
Minimalism is key. Nothing in the trait body is relevant to this topic.
```
#[multiversx_sc::contract]
#[esdt_attribute("testBasic", BigUint)]
pub trait SomeContract {
}
```

is enough

Reply:
Also, let's not make it the adder. Just give it some name. This is not about a specific contract